platform/backend/src/mcp-server-runtime/manager.ts

```diff
@@ -40,36 +113,47 @@ export class McpServerRuntimeManager {
   constructor() {
     this.k8sConfig = new k8s.KubeConfig();
 
-    // Load K8s config from environment or default locations
+    const hasCustomKubeconfig =
+      Boolean(kubeconfig && kubeconfig.trim().length > 0);
+
     try {
+      // If not loading from cluster AND user provided a custom kubeconfig → validate it
+      if (!loadKubeconfigFromCurrentCluster && hasCustomKubeconfig) {
+        validateKubeconfig(kubeconfig!);
+      }
+
       if (loadKubeconfigFromCurrentCluster) {
-        /**
-         * Running inside a K8s cluster
-         *
-         * Automatically configure the client to connect to the Kubernetes API
-         * when the application is running inside a Kubernetes cluster
-         */
+        // Running inside Kubernetes pod
         this.k8sConfig.loadFromCluster();
-      } else if (kubeconfig) {
-        // Load from KUBECONFIG env var
-        this.k8sConfig.loadFromFile(kubeconfig);
+        logger.info("Loaded kubeconfig from in-cluster environment");
+      } else if (hasCustomKubeconfig) {
+        // Custom kubeconfig path provided by developer
+        logger.info(`Loading kubeconfig from env var: ${kubeconfig}`);
+        this.k8sConfig.loadFromFile(kubeconfig!);
       } else {
-        // Load from default location (~/.kube/config)
+
+        // No env var? → Use ~/.kube/config (default Kubernetes path)
+        logger.info("No kubeconfig env set — using default kubeconfig (~/.kube/config)");
         this.k8sConfig.loadFromDefault();
       }
 
-      // Only create API client if K8s config loaded successfully
+      // Initialize Kubernetes API client
       this.k8sApi = this.k8sConfig.makeApiClient(k8s.CoreV1Api);
+
+      this.k8sAttach = new Attach(this.k8sConfig);
+      this.k8sLog = new k8s.Log(this.k8sConfig);
+      this.namespace = namespace;
     } catch (error) {
-      logger.error({ err: error }, "Failed to load Kubernetes config:");
+      logger.error({ err: error }, "Failed to load Kubernetes config");
       this.status = "error";
-    }
+      this.k8sAttach = undefined as any;
```

Comment:
Changing the `this.k8sAttach`'s type to `Attach | undefined` would reflect reality better. Also avoid typecasting. Same for the other 2 vars.